pkg/spire/controller.go

```diff
@@ -45,14 +47,30 @@ func init() {
 // controllerKey is a way to associate the ControllerAPIClient from inside the context.Context
 type controllerKey struct{}
 
+// OnStore stores the changed spire config into the SpireClientApi
+func OnStore(ctx context.Context, logger *zap.SugaredLogger) func(name string, value interface{}) {
```

Comment:
question: why do we need to implement OnStore? Are we supporting changing the SPIRE client without a controller restart if the config changes?

Reply:
The configStore reads in the config and stores it into its local structures. Spire controller maintains its own local config. This does not get updated with the changes in the actual configMap. Essentially the Spire ConfigMap does not make it to the Spire Controller without this callback.

One way to get around this without the callback could be to load from the ConfigStore into the SpireCntroller at the beginning of the reconciler as done for other configs whcih get copied into the. context.

https://github.com/tektoncd/pipeline/blob/8f76a04852ed947bab874d43eb052d2ea8cabbe2/pkg/client/injection/reconciler/pipeline/v1beta1/taskrun/reconciler.go#L190

which will result in something like this.
	// If configStore is set, attach the frozen configuration to the context.
	if r.configStore != nil {
		ctx = r.configStore.ToContext(ctx)
		r.SpireClient.SetConfig(ctx)
	}

WDYT?